price-feeder/oracle/provider/okx.go

```diff
@@ -0,0 +1,212 @@
+	"github.com/gorilla/websocket"
+	"github.com/umee-network/umee/price-feeder/oracle/types"
+)
+
+const (
+	okxHost          = "ws.okx.com:8443"
+	okxPath          = "/ws/v5/public"
+	msReadNewMessage = 50
+)
+
+var _ Provider = (*OkxProvider)(nil)
+
+type (
+	// OkxProvider defines an Oracle provider implemented by the Okx public
+	// API.
+	//
+	// REF: https://www.okx.com/docs-v5/en/#websocket-api-public-channel-tickers-channel
+	OkxProvider struct {
+		wsURL            url.URL
+		wsClient         *websocket.Conn
+		tickersMap       *sync.Map // InstId => OkxTickerPair
+		msReadNewMessage uint16
+	}
+
+	// OkxTickerPair defines a ticker pair of Okx
+	OkxTickerPair struct {
+		InstId string `json:"instId"` // Instrument ID ex.: BTC-USDT
+		Last   string `json:"last"`   // Last traded price ex.: 43508.9
+		Vol24h string `json:"vol24h"` // 24h trading volume ex.: 11159.87127845
+	}
+
+	// OkxTickerResponseWS defines the response structure of a Okx ticker
+	// request.
+	OkxTickerResponseWS struct {
+		Data []OkxTickerPair   `json:"data"`
+		Arg  SubscriptionTopic `json:"arg"`
+	}
+
+	SubscriptionTopic struct {
+		Channel string `json:"channel"` // Channel name ex.: tickers
+		InstId  string `json:"instId"`  // Instrument ID ex.: BTC-USDT
+	}
+
+	SubscriptionMsg struct {
+		Op   string              `json:"op"` // Operation ex.: subscribe
+		Args []SubscriptionTopic `json:"args"`
+	}
+)
+
+// NewOkxProvider creates a new OkxProvider
+func NewOkxProvider(ctx context.Context, pairs ...types.CurrencyPair) (*OkxProvider, error) {
+	wsURL := url.URL{
+		Scheme: "wss",
+		Host:   okxHost,
+		Path:   okxPath,
+	}
+
+	wsConn, _, err := websocket.DefaultDialer.Dial(wsURL.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to ws: %+v", err)
+	}
+
+	p := &OkxProvider{
+		wsURL:            wsURL,
+		wsClient:         wsConn,
+		tickersMap:       &sync.Map{},
+		msReadNewMessage: msReadNewMessage,
+	}
+
+	if err := p.newTickerSubscription(pairs...); err != nil {
+		return nil, err
+	}
+
+	go p.handleTickers(ctx)
+
+	return p, nil
+}
+
+// GetTickerPrices returns the tickerPrices based on the saved map
+func (p OkxProvider) GetTickerPrices(pairs ...types.CurrencyPair) (map[string]TickerPrice, error) {
+	tickerPrices := make(map[string]TickerPrice, len(pairs))
+
+	for _, currencyPair := range pairs {
+		cp := currencyPair
+		price, err := p.getTickerPrice(cp)
+		if err != nil {
+			return nil, err
+		}
+
+		tickerPrices[cp.String()] = price
+	}
+
+	return tickerPrices, nil
+}
+
+func (p OkxProvider) getTickerPrice(cp types.CurrencyPair) (TickerPrice, error) {
+	instrumentId := getInstrumentId(cp)
+	tickerPair, exist := p.getMapTicker(instrumentId)
+	if !exist { // it should exist the ticker will be
+		return TickerPrice{}, fmt.Errorf("ticker pair not found %s", instrumentId)
+	}
+
+	return tickerPair.ToTickerPrice()
+}
+
+func (p OkxProvider) getMapTicker(instrumentId string) (pair OkxTickerPair, exists bool) {
```

Comment:
Should return an error instead, and probably use unnamed return vars since the other functions do : 
```suggestion
func (p OkxProvider) getMapTicker(instrumentId string) (OkxTickerPair, error) {
```